piccolo_admin/endpoints.py

```diff
@@ -248,10 +248,30 @@ def __init__(
         table_configs: t.List[TableConfig] = []
 
         for table in tables:
+            _table = None
             if isinstance(table, TableConfig):
-                table_configs.append(table)
+                _table = table
             else:
-                table_configs.append(TableConfig(table_class=table))
+                _table = TableConfig(table_class=table)
+
+            try:
+                columns = _table.get_visible_columns()
+                for column in columns:
+                    if column._meta.secret and column._meta.required:
+                        print(
+                            """
+WARNING: %s.%s is using `secret` and `required` column kwargs which are incompatible.
+You may encounter unexpected behavior when using this table within Piccolo Admin.
+                            """
+                            % (
+                                _table.table_class._meta.tablename,
+                                column._meta._name,
+                            )
+                        )
+            except Exception as e:
+                pass
```

Comment:
@ethagnawl @dantownsend Or we can use `warnings.warn` and show `UserWarning` in terminal after starting server like this
```python
import warnings
# remove all previous changes in PR code and add this code at line 273
for table_config in table_configs:
    table_class = table_config.table_class
    for column in table_class._meta.columns:
        if column._meta.secret and column._meta.required:
            message = (
                f"{table_class._meta.tablename}."
                f"{column._meta._name} is using `secret` and "
                f"`required` column args which are incompatible. "
                f"You may encounter unexpected behavior when using "
                f"this table within Piccolo Admin."
            )
            warnings.warn(message)
```

Reply:
@ethagnawl Thanks a lot for this. It's always tricky in these situation to know what's best - warnings, print statements, logging, raise an exception, or even `sys.exit`.

@sinisaos I don't really have a strong preference. Piccolo has `colored_warning` too:

https://github.com/piccolo-orm/piccolo/blob/f2a2fb8cd91e29b1d0b90b3a8e01a43f2ae502dd/piccolo/utils/warnings.py#L22

Reply:
@dantownsend Similar way with `colored_warning`
```python
from piccolo.utils.warnings import Level, colored_warning
# remove all previous changes in PR code and add this code at line 273
for table_config in table_configs:
    table_class = table_config.table_class
    for column in table_class._meta.columns:
        if column._meta.secret and column._meta.required:
            message = (
                f"{table_class._meta.tablename}."
                f"{column._meta._name} is using `secret` and "
                f"`required` column args which are incompatible. "
                f"You may encounter unexpected behavior when using "
                f"this table within Piccolo Admin."
            )
            colored_warning(message, level=Level.high)
```
This seems better to me, but as you and @ethagnawl decide, it will be fine. I agree with both ways.